src/components/WeatherDetails.js

```diff
@@ -0,0 +1,38 @@
+import { useSelector } from 'react-redux';
+import { useParams, Link } from 'react-router-dom';
+import AqiDetails from './AqiDetails';
+import AqiDetailed from './AqiDetailed';
+import './styles/WeatherDetails.css';
+
+const WeatherDetails = () => {
+  const cityData = useSelector((state) => state.weather.dataOfCities);
+  const { id } = useParams();
+  const selectedCityData = cityData.find((city) => city.id === id);
+  const cityName = selectedCityData ? selectedCityData.city : '';
+  const Aqi = AqiDetails(id, cityData);
```

Comment:
- [ ] The same happens here, instead of `Aqi` it would be better to keep it lower case since it represents an array of data. 🤓 🔧 

```suggestion
  const aqi = aqiDetails(id, cityData);
```

Reply:
Implemented